nemo/collections/tts/models/fastpitch.py

```diff
@@ -142,16 +145,22 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         speaker_emb_condition_prosody = cfg.get("speaker_emb_condition_prosody", False)
         speaker_emb_condition_decoder = cfg.get("speaker_emb_condition_decoder", False)
         speaker_emb_condition_aligner = cfg.get("speaker_emb_condition_aligner", False)
+        energy_embedding_kernel_size = cfg.get("energy_embedding_kernel_size", 0)
+        energy_predictor = None
+        if self._cfg.get("energy_predictor", None) is not None:
+            energy_predictor = instantiate(self._cfg.energy_predictor)
```

Comment:
copy and paste my comments from the closed PR:https://github.com/NVIDIA/NeMo/pull/5176

> But energy_predictor needs to be initialized to None. Because energy_predictor in line 150 is variable for the Neural module for predicting energy. Whereas, self._cfg.get("energy_predictor", None) will give us just the config, it is only when we call instantiate in line 152 that energy_predictor gets it's value (the neural module).

It is still not necessary because if the input cfg is `None` then `instantiate` func will directly return `None` (ref: https://github.com/facebookresearch/hydra/blob/main/hydra/_internal/instantiate/_instantiate2.py#L178-L179). So in this case, you only need,

```python
energy_predictor = instantiate(self._cfg.get("energy_predictor", None))
```

Reply:
Done.